packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -0,0 +1,381 @@
+        Artifacts.setUp();
+
+        string memory path = string.concat(vm.projectRoot(), "/deploy-config/", deploymentContext, ".json");
+        vm.etch(address(cfg), vm.getDeployedCode("DeployConfig.s.sol:DeployConfig"));
+        vm.label(address(cfg), "DeployConfig");
+        vm.allowCheatcodes(address(cfg));
+        cfg.read(path);
+
+        outfile = string.concat(vm.projectRoot(), "/deployments/", deploymentContext, "/genesis-l2.json");
+
+        _loadAddresses(string.concat(vm.projectRoot(), "/deployments/", deploymentContext, "/.deploy"));
+    }
+
+    function run() public {
+        _setPrecompiles();
+        _setProxies();
+        _setImplementations();
+
+        /// Reset so its not included state dump
+        vm.etch(address(cfg), "");
+
+        vm.dumpState(outfile);
+        _sortJsonByKeys(outfile);
+    }
+
+    /// @notice Give all of the precompiles 1 wei so that they are
+    ///         not considered empty accounts.
+    function _setPrecompiles() internal {
+        for (uint256 i; i < PRECOMPILE_COUNT; i++) {
+            vm.deal(address(uint160(i)), 1);
+        }
+    }
+
+    /// @dev Set up the accounts that correspond to the predeploys.
+    ///      The Proxy bytecode should be set. All proxied predeploys should have
+    ///      the 1967 admin slot set to the ProxyAdmin predeploy. All defined predeploys
+    ///      should have their implementations set.
+    function _setProxies() internal {
+        bytes memory code = vm.getDeployedCode("Proxy.sol:Proxy");
+        uint160 prefix = uint160(0x420) << 148;
+        for (uint256 i = 0; i < PROXY_COUNT; i++) {
+            address addr = address(prefix | uint160(i));
+            if (_notProxied(addr)) {
+                continue;
+            }
+
+            vm.etch(addr, code);
+            vm.store(addr, PROXY_ADMIN_ADDRESS, bytes32(uint256(uint160(Predeploys.PROXY_ADMIN))));
+
+            if (_isDefinedPredeploy(addr)) {
+                address implementation = _predeployToCodeNamespace(addr);
+                vm.store(addr, PROXY_IMPLEMENTATION_ADDRESS, bytes32(uint256(uint160(implementation))));
+            }
+        }
+    }
+
+    /// @notice LEGACY_ERC20_ETH is not being predeployed
+    function _setImplementations() internal {
+        _setLegacyMessagePasser();
+        _setDeployerWhitelist();
+        _setWETH9();
+        _setL2StandardBridge();
+        _setL2CrossDomainMessenger();
+        _setSequencerFeeVault();
+        _setOptimismMintableERC20Factory();
+        _setL1BlockNumber();
+        _setGasPriceOracle();
+        _setGovernanceToken();
+        _setL1Block();
+    }
+
+    function _setLegacyMessagePasser() internal {
+        address impl = _predeployToCodeNamespace(Predeploys.LEGACY_MESSAGE_PASSER);
+        vm.etch(
+            impl,
+            vm.getDeployedCode("LegacyMessagePasser.sol:LegacyMessagePasser")
+        );
+
+        _verifyProxyImplementationAddress(Predeploys.LEGACY_MESSAGE_PASSER, impl);
```

Comment:
Can we break the post checks out into specific functions for each contract? ie

`_checkLegacyMessagePasser`

Then we can either: 1) have a single `postCheck` that calls all checks or 2) keep checks in the set functions. I have a slight preference for 2) but 1) is probably easier to maintain/extend so fine with either solution